### PR TITLE
Connection quality

### PR DESCRIPTION
**Pull request by hiroshihorie**

### CODE REVIEW

File: Sources/LiveKit/participant/RemoteParticipant.swift [hunk 1]
```diff
@@ -4,8 +4,9 @@ import Promises
 
 public class RemoteParticipant: Participant {
 
-    init(sid: String, info: Livekit_ParticipantInfo?) {
+    init(sid: String, info: Livekit_ParticipantInfo?, room: Room) {
         super.init(sid: sid)
+        self.room = room
```
Comment (boks1971) [inline Sources/LiveKit/participant/RemoteParticipant.swift:9]: Is this just for consistency in `init` signature with `LocalParticipant` as this new field does not seem to be used anywhere OR is it some interface/implementation making it necessary to change this signature?

**Review by boks1971 — APPROVED** [review]
lgtm! One small question.

File: Sources/LiveKit/participant/RemoteParticipant.swift (hunk 1 above)
Comment (hiroshihorie) [reply Sources/LiveKit/participant/RemoteParticipant.swift:9]: It's actually used in the parent Participant class, which I forgot to do before! So I fixed it now 😅

**Comment by hiroshihorie** [comment]
Thanks Raja oniichaannn !! 🙏

File: Sources/LiveKit/participant/Participant.swift [hunk 2]
```diff
@@ -8,18 +8,25 @@ public class Participant: MulticastDelegate<ParticipantDelegate> {
...
+            notify { $0.participant(self, didUpdate: self.metadata) }
+            room?.notify { $0.room(self.room!, participant: self, didUpdate: self.metadata) }
+        }
+    }
+
+    public internal(set) var connectionQuality: ConnectionQuality = .unknown {
+        didSet {
+            guard oldValue != connectionQuality else { return }
+            notify { $0.participant(self, didUpdate: self.connectionQuality) }
+            room?.notify { $0.room(self.room!, participant: self, didUpdate: self.connectionQuality) }
```
Comment (hiroshihorie) [inline Sources/LiveKit/participant/Participant.swift:29]: `room` is used here 😄

File: Sources/LiveKit/participant/RemoteParticipant.swift (hunk 1 above)
Comment (boks1971) [reply Sources/LiveKit/participant/RemoteParticipant.swift:9]: Got it. Thank you!

**Pull request merged**